Cohort3/menu categorias

Edu-Fortes opened the pull request:
Menu das categorias para Desktop e Mobile.
![menu__360px](https://user-images.githubusercontent.com/46508709/213921785-08263c9f-b9a9-4bfd-b8cd-bb5832a95f3b.JPG)
![menu__1440px](https://user-images.githubusercontent.com/46508709/213921789-ba368fd3-2444-4b8e-a188-48f2a4548b91.JPG)

Edu-Fortes committed:
Initial commit, add images folder

Edu-Fortes committed:
Change to mobile first

Edu-Fortes committed:
Finish mobile first aproach to menu

Edu-Fortes committed:
Add max-width to BODY thru PAGE.CSS

thiagopuntar commented on src/blocks/page/page.css:
```diff
@@ -0,0 +1,6 @@
+  max-width: 1440px;
+  margin: 0 auto;
+  font-family: "Inter", Arial, sans-serif;
+  background: linear-gradient(180deg, #0a2a34 0%, rgba(80, 138, 156, 0.2) 100%);
```
Adicione um atributo fixed no background em telas maiores, o comportamento do linear-gradient é começar a repetir assim que acabar o tamanho da tela.

Edu-Fortes replied:
Não me atentei a essa possibilidade. Testei colocando um container para criar rolagem de página e visualizei o problema de repetição do background.

thiagopuntar commented on src/blocks/menu/menu.css:
```diff
@@ -0,0 +1,9 @@
+.menu {
+  margin: 2.5rem 1.3rem;
```
Esses elementos nas bordas do body geralmente dão um pouco de dor de cabeça pra gente setar margin (tanto no header quando no footer). Acho que você pode trocar essas margins (pelo menos as verticais) por um padding no container. No caso específico, reparei que ao a gente adicionar essa margin no elemento do top, acabou aumentando tamanho do body. Não acho que tenha necessidade da gente ter esse scroll por conta do margin.

Edu-Fortes replied:
Concordo, ficou bem melhor aplicando as alterações apontadas.

thiagopuntar commented on src/blocks/menu/__image/menu__image.css:
```diff
@@ -0,0 +1,3 @@
+.menu__image {
+  max-height: inherit;
```
max-height é uma prop que geralmente a gente quer evitar de utilizar. Ela pode causar uns comportamentos indesejados como por exemplo overflow. 
No caso, eu tirei as props rodando local e continuou a funcionar normalmente. Tente remover todos os max-heights pra ver como fica

Edu-Fortes replied:
Realmente tirando o max-height não latera a disposição. Mas estava querendo seguir o Figma e na versão mobile o ícone fica menor. Alterei utilizando um scale, me fale o que acha.

thiagopuntar commented on src/blocks/menu/__paragraph/menu__paragraph.css:
```diff
@@ -0,0 +1,13 @@
+  font-size: 0.8rem;
+  font-weight: 700;
+  line-height: 1.3;
+  color: hsla(0, 0%, 67%, 1);
```
Excelente utilizar hsla, muito obrigado meu amigo.
Vamos tentar fazer algo a mais? Que tal isolar as cores em um css de theme como uma variável css?
Qualquer dúvida me fala

Edu-Fortes replied:
Nossa, sensacional trabalhar com variável nesse tipo de customização. Obrigado pelo apontamento, não havia pensado nessa possibilidade.

thiagopuntar commented on src/index.html:
```diff
@@ -1,13 +1,42 @@
+  <body class="page">
+    <nav class="menu">
+      <ul class="menu__bar">
+        <li class="menu__item">
```
Senti falta de um button dentro desse li, só por conta do que representa esses menu semanticamente. Quando a gente clicar nos elementos, queremos desempenhar ações. Excelente role para um button.

Edu-Fortes replied:
Perfeito. Faz mais sentido mesmo adicionar um button.

Edu-Fortes committed:
Add modification sugested by Puntar, rev1

thiagopuntar commented on src/blocks/menu/__image/menu__image.css:
```diff
@@ -1,3 +1,9 @@
 .menu__image {
-  max-height: inherit;
+  transform: scale(0.75);
```
Fala Edu, desculpa a demora! E desculpa sobre o height, não tinha reparado nessa questão que os ícones tinham outro tamanho no mobile. Acredito que você possa usar o height então, nesse caso que não tem uma proporção, pode ficar a vontade de usar o height em px para uma img. A gente só recomenda ter cautela de utilizar height e max-height em seções da página que pode acabar causando overflow, mas no caso de uma imagem acho que funciona de boa... Talvez, na verdade, só de setar um width na imagem, pelo aspect ratio natural, a height já acompanhe o tamanho setado na width. Tenta fazer o teste.
Acho que dessa forma é um pouco melhor que o transform.. O transform você vai utilizar ao selecionar a categoria, provavelmente.

Edu-Fortes replied:
Opa Puntar, imagina da demora! Estávamos de férias. Fiz o teste utilizando o width para não precisar usar o height, mas não da muito certo por conta da proporção da batata frita, fica muito diferente em relação aos outros ícones. No final utilizei o height mesmo.  Removi o max-height da classe menu__item, esse não estava ajudando em nada.

thiagopuntar reviewed (commented):
Fala Edu, ficou excelente o trabalho! Parabéns, desculpa pelo height, não tinha me atentado. Tenta fazer os testes que eu comentei e a gente já aprova na sequencia.

Ainda não revisei a do hover porque vou esperar mergear essa para ver como ficou por lá, blz?

Edu-Fortes committed:
fix(menu): change transform: scale from menu__image to max-height

As Puntar suggestion, tried to set max-widht in menu-image but ho help due french fries proportions. In the end of the day used max-height to maintain proportion on all images
